lib/utils/market-hours.ts

```
@@ -0,0 +1,240 @@
+ * After-hours: 4:00 PM - 8:00 PM ET
+ *
+ * Holidays are calculated dynamically for any year using known US holiday rules.
+ * No hardcoded year-specific data needed.
+ */
+
+// ── Holiday Calculator ──
+
+/** Nth weekday of a month. weekday: 0=Sun...6=Sat */
+function nthWeekday(year: number, month: number, weekday: number, n: number): Date {
+    const first = new Date(Date.UTC(year, month - 1, 1));
+    let day = first.getUTCDay(); // 0=Sun
```

ESLint

[error] 'day' is never reassigned. Use 'const' instead.

(prefer-const)